backend/src/common/util/template_util.go

```diff
@@ -56,5 +57,14 @@ func ValidateWorkflow(template []byte) (*v1alpha1.Workflow, error) {
 	if wf.Kind != argoK8sResource {
 		return nil, NewInvalidInputError("Unexpected resource type. Expected: %v. Received: %v", argoK8sResource, wf.Kind)
 	}
+	if wf.Spec.WorkflowTemplateRef != nil {
```

Comment:
KFP never uses workflowtemplateref, I think we don't want to validate it.

Reply:
Ohh, it should actually be the opposite in the `if` statement but I guess we could drop it as well.